app/core/component/forms/forms.controller.js

```diff
@@ -2,50 +2,42 @@
 
 angular
     .module('forms')
-    .controller('formsController', function(
-        $rootScope, addCourseEvent, pushCourseToEditFormEvent,
-        editCourseEvent, toggleVisibilityFromForAddEvent
-    ) {
+    .controller('formsController', function($rootScope, courseService, eventsFactory) {
         // eslint-disable-next-line consistent-this,no-invalid-this
         var $ctrl = this;
 
-        $rootScope.$on(toggleVisibilityFromForAddEvent, function(event, data) {
+        $rootScope.$on(eventsFactory.toggleVisibilityFromForAddEvent, function(event, data) {
             $ctrl.showFromForAdd = data;
         });
 
         $ctrl.addCourse = function() {
             var course = {
                 title: $ctrl.courseTitle,
                 description: $ctrl.courseDescription,
-                uploadDate: new Date().toISOString()
+                duration: $ctrl.courseDuration,
+                creationDate: new Date().toISOString()
             };
-
-            $rootScope.$broadcast(addCourseEvent, course);
+            courseService.addCourse(course);
             $ctrl.courseTitle = '';
             $ctrl.courseDescription = '';
+            $ctrl.courseDuration = '';
             $ctrl.showFromForAdd = !$ctrl.showFromForAdd;
         };
 
-        $rootScope.$on(pushCourseToEditFormEvent, function(event, data) {
-            $ctrl.selectedCourse = data;
-            $ctrl.courseTitleToEdit = data.title;
-            $ctrl.courseDescriptionToEdit = data.description;
-            $ctrl.courseUploadDateToEdit = data.uploadDate;
-            $ctrl.courseDurationToEdit = data.duration;
+        $rootScope.$on(eventsFactory.pushCourseToEditFormEvent, function(event, data) {
```

Comment:
pushCourseToEditFormEvent says that you should push Course:) 
so I believe that data should be renamed.

Reply:
renamed;
courseExchangeWithEditForm is ok for it?

Reply:
sure, thanks